code/modules/clothing/head/misc_special.dm

```diff
@@ -67,23 +66,20 @@
 	if (istype(location, /turf))
 		location.hotspot_expose(700, 1)
 
-/obj/item/clothing/head/cakehat/attack_self(mob/user)
-	if(status > 1)	return
-	src.onfire = !( src.onfire )
-	if (src.onfire)
-		src.force = 15
-		src.AddLuminosity(brightness_on)
-		src.damtype = "fire"
-		src.icon_state = "cake1"
-		SSobj.processing |= src
-	else
-		src.force = null
-		src.AddLuminosity(-brightness_on)
-		src.damtype = "brute"
-		src.icon_state = "cake0"
-	return
+/obj/item/clothing/head/hardhat/cakehat/turn_on()
+    ..()
+    force = 15
+    damtype = BURN
+    SSobj.processing |= src
 
+/obj/item/clothing/head/hardhat/cakehat/turn_off()
+    ..()
+    force = 0
+    damtype = BRUTE
+    SSobj.processing -= src
 
+/obj/item/clothing/head/hardhat/cakehat/is_hot()
+ 	return on * heat

```

Comment:
You indented with spaces from line 70 onward. Make sure you use tabs. I must've missed this earlier, sorry.